fakestorage/upload.go

```diff
@@ -203,9 +203,19 @@ func (s *Server) checkUploadPreconditions(r *http.Request, bucketName string, ob
 	return nil
 }
 
+func normalizeFileName(name string) string {
+	for {
+		name = strings.Replace(name, "/", "", 1)
+		if !strings.HasPrefix(name, "/") {
+			break
+		}
+	}
+	return name
```

Comment:
Could use strings.TrimLeft. I don't think this is what we want though. Like, doesn't GCS treat `file.txt` and `/file.txt` as different objects?

Reply:
@fsouza it is the same objects

Reply:
Changed to `strings.TrimLeft`

Reply:
Some libraries even remove leading slashes from own side - https://github.com/googleapis/google-cloud-node/pull/2269/files

Reply:
Fixed in different way - https://github.com/fsouza/fake-gcs-server/pull/918/commits/41e8db7594af4433bf6b0818ef463df0764cd16d